cluster/main.tf

```diff
@@ -13,16 +13,35 @@ data "aws_availability_zones" "available" {
   state = "available"
 }
 
+data "aws_region" "current" {}
+
 data "aws_ami" "kubernetes_ami" {
   most_recent = true
 
   # ebs-kubernetes-baseimage-1.7-201712051032
   # This is a pre build image base on https://github.com/skyscrapers/kubernetes-baseimage
-  name_regex = "^ebs-kubernetes-baseimage-${element(split(".",var.k8s_version),0)}.${element(split(".",var.k8s_version),1)}-*"
+  name_regex = "^ebs-kubernetes-baseimage-${local.k8s_short_version}-*"
 
   owners = ["496014204152"]
 }
 
+resource "aws_ami_copy" "k8s_base_image" {
+  count = "${var.k8s_image_encryption ? 1 : 0}"
+
+  # For now, this resource will replace the copied AMI everytime there's a new source AMI, so there'll always be just one AMI in the target account. Until this is implemented: https://github.com/terraform-providers/terraform-provider-aws/issues/792
+
+  name              = "${data.aws_ami.kubernetes_ami.name}"
+  description       = "Kubernetes base image"
+  source_ami_id     = "${data.aws_ami.kubernetes_ami.id}"
+  source_ami_region = "${data.aws_region.current.name}"
+  encrypted         = true
```

Comment:
I think we should also set the kms key id (if set) for the encryption of the base ami